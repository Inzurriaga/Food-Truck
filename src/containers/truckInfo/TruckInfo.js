import React, { Component } from "react"
import { connect } from "react-redux"
import { getTruckInfo } from "../../thunk/GetTruckInfo"

export class TruckInfo extends Component{
    constructor(){
        super();
        this.state = {
            truckInfo: {}
        }
    }

    componentDidMount(){
        this.truckInfo(this.props.id)
    }

    truckInfo =  async(id) => {
        const url = `http://localhost:3000/api/${id}`
        const truckInfo = await this.props.getTruckInfo(url)
        this.setState({ truckInfo })
    }
    
    render(){
        const {name, site, descrip } = this.state.truckInfo;
        return(
            <section className="truck-info-container">
                <article>
                    <div className="picture">
                        picture of the truck
                    </div>
                    <div className="info">
                        <a href={`${site}`}>{site}</a>
                    </div>
                    <div className="description">
                        <h2>{name}</h2>
                        <p>{descrip}</p>
                    </div>
                    <div className="menu">
                        <h2>Menu</h2>
                    </div>
                </article>
            </section>
        )
    }
}

export const mapDispatchToProps = (dispatch) => ({
    getTruckInfo: url => dispatch(getTruckInfo(url))
})

export default connect(null, mapDispatchToProps)(TruckInfo)