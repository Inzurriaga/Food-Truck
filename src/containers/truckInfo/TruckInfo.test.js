import React from "react";
import { shallow } from "enzyme";
import TruckInfo from "./TruckInfo"

describe("TruckInfo", () => {
    describe("TruckInfo", () => {
        let wrapper;
        beforeEach(() => {
            wrapper = shallow(
                <TruckInfo />
            )
        })
        it("should match snap shot", () => {
            expect(wrapper).toMatchSnapshot()
        })
    })
})