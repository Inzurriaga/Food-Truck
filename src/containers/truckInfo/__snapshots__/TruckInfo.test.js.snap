// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TruckInfo TruckInfo should match snap shot 1`] = `
<section
  className="truck-info-container"
>
  <article>
    <div
      className="picture"
    >
      picture of the truck
    </div>
    <div
      className="info"
    >
      truck type like cuisine rating
    </div>
    <div
      className="description"
    >
      truck description
    </div>
    <div
      className="menu"
    >
      menu
    </div>
  </article>
</section>
`;
